faucet/components/Faucet.tsx

```diff
@@ -0,0 +1,121 @@
+import { useState, useEffect } from 'react'
+import { isAddress } from 'web3-utils'
+import type { Networks } from '@hoprnet/hopr-ethereum/scripts/addresses'
+
+async function mint(
+  network: Networks,
+  address: string
+): Promise<{
+  success: boolean
+  transactionHash?: string
+  message?: string
+}> {
+  const response = await fetch(`/api/mint?network=${network}&address=${address}`)
+
+  if (response.status === 200) {
+    const { transactionHash } = await response.json()
+    return {
+      success: true,
+      transactionHash,
+    }
+  } else {
+    return {
+      success: false,
+      message: response.statusText,
+    }
+  }
+}
+
+function Faucet({ network }: { network: Networks }) {
+  const [address, setAddress] = useState<string>(undefined)
+  const [isValidAddress, setIsValidAddress] = useState<boolean>(undefined)
+  const [status, setStatus] = useState<'PENDING' | 'SUCCESS' | 'FAILURE'>(undefined)
+  const [message, setMessage] = useState<string>(undefined)
+  const isButtonDisabled = !isValidAddress || status === 'PENDING'
+  const showStatusMessage = status === 'SUCCESS' || status === 'FAILURE'
+
+  // update 'isValidAddress' on 'address' change
+  useEffect(() => {
+    if (typeof address === 'undefined' || address === '') {
+      setIsValidAddress(undefined)
+    } else {
+      setIsValidAddress(isAddress(address))
+    }
+  }, [address])
+
+  // once button is clicked
+  const onClick = async () => {
+    setStatus('PENDING')
+
+    const result = await mint(network, address)
+
+    if (result.success) {
+      setMessage(result.transactionHash)
+      setStatus('SUCCESS')
+    } else {
+      setMessage(result.message)
+      setStatus('FAILURE')
+    }
+  }
+
+  return (
+    <div className="container">
+      <div className="inputs">
+        <input onChange={(e) => setAddress(e.target.value)} defaultValue={address} />
+        &nbsp;
+        <button onClick={onClick} disabled={isButtonDisabled}>
+          Give me hopr tokens!
+        </button>
+      </div>
+      {!showStatusMessage ? null : status === 'SUCCESS' ? (
```

Comment:
This looks a bit hard to read IMHO. We check that the `status` is successful, yet we display a message that the transaction is pending? Perhaps we can distinguish between a “successful” minting (i.e. a published transaction) and an actual successful minting (i.e. a confirmed transaction).

We might as well do the following in the sake of readability:
```
{ showStatusMessage &&
   status === 'SUCCESS' ?
     // Display object for “successful” tx
    :
     // Display the something broke message
}
```